src/app.md

```diff
@@ -189,6 +190,85 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+[ex-tar-decompress-remove-path-prefix]: #ex-tar-decompress-remove-path-prefix
+<a name="ex-tar-decompress-remove-path-prefix"></a>
+## Decompress a tarball while removing a prefix from the paths
+
+[![flate2-badge]][flate2] [![tar-badge]][tar] [![cat-compression-badge]][cat-compression]
+
+Decompresses the tarball by iterating over the entries and unpacking them after
+removing a specified path prefix.
+
+Create a [`tar::Archive`] from a [`flate2::read::GzDecoder::new`], iterate over
+its [`tar::Entries`], and for each [`tar::Entry`] we [`tar::Entry::unpack`] it
+to a given path.  The paths are constructed by getting the [`tar::Entry::path`]
+for each entry and then calling the [`std::path::Path::strip_prefix`] method, to
+strip the specified prefix (`bundle/logs` in this case) from the path. Note that
+because maps are lazily evaluated we must iterate (or use some other method to
+exhaust the iterator) over the map to actually unpack the archive.
+
+```rust,no_run
+# #[macro_use]
+# extern crate error_chain;
+
+extern crate flate2;
+extern crate tar;
+
+#
+# error_chain! {
+#     foreign_links {
+#         Io(::std::io::Error);
+#         StripPrefixError(::std::path::StripPrefixError);
+#     }
+# }
+
+use std::fs::File;
+use std::path::PathBuf;
+use flate2::read::GzDecoder;
+use tar::Archive;
+
+fn run() -> Result<()> {
+    // Get a reference to archive.tar.gz, which is a sample tarball created for
+    // this example
+    let file = File::open("archive.tar.gz")?;
+
+    // First get a streaming decoder to handle the gzip decompression and pass
+    // that into a tar::Archive
+    let mut archive = Archive::new(GzDecoder::new(file)?);
+
+    // Specify prefix we are removing
+    let prefix = "bundle/logs";
+
+    let extracted_entries = archive
```

Comment:
`extracted_entries` might not bethe best name as these will not get extracted until you consume this iterator. 
`entries` will suffice